fast/stages/3-network-security/variables.tf

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "factories_config" {
+  description = "Configuration for network resource factories."
+  type = object({
+    data_dir = optional(string, "data")
+  })
+  default  = {}
+  nullable = false
+  validation {
+    condition     = var.factories_config.data_dir != null
+    error_message = "Data folder needs to be non-null."
+  }
+}
+
+variable "ngfw_enterprise_config" {
+  description = "NGFW Enterprise configuration."
+  type = object({
+    endpoint_zones   = optional(list(string), ["europe-west1-b", "europe-west1-c", "europe-west1-d"])
```

Comment:
move the regions to the default, not as optional default

plus, I would use primary/secondary and inherit region mappings from the networking stage output file

Reply:
This is by design. There are two reasons I don't do this:
* IHAC where endpoints are not activate in all zones/regions automatically for different reasons, so I'd like to keep the two things (where workloads are and where endpoints are) separata
* inheir regions from the networking stage would imply that I create components based on the fact that there are one or two regions, which makes everything terribly complex

Reply:
My point is different: why do you want to hardwire regions in this stage, and then force users to change them, when that information is already present in the network stage, and you can infer it from there?